moveit_ros/planning_interface/test/subframes_test.cpp

```diff
@@ -0,0 +1,198 @@
+  cylinder.primitive_poses[0].position.x = 0.0;
+  cylinder.primitive_poses[0].position.y = 0.0;
+  cylinder.primitive_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, 90.0 / 180.0 * M_PI, 0);
+  cylinder.primitive_poses[0].orientation = tf2::toMsg(orientation);
+
+  cylinder.subframe_poses.resize(1);
+  cylinder.subframe_names.resize(1);
+  cylinder.subframe_names[0] = "tip";
+  cylinder.subframe_poses[0].position.x = 0.03;
+  cylinder.subframe_poses[0].position.y = 0.0;
+  cylinder.subframe_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, 90.0 / 180.0 * M_PI, 0);
+  cylinder.subframe_poses[0].orientation = tf2::toMsg(orientation);
+
+  // Lastly, the objects are published to the PlanningScene. In this tutorial, we publish a box and a cylinder.
+  box.operation = moveit_msgs::CollisionObject::ADD;
+  cylinder.operation = moveit_msgs::CollisionObject::ADD;
+  planning_scene_interface.applyCollisionObjects({ box, cylinder });
+}
+
+TEST(TestPlanUsingSubframes, SubframesTests)
+{
+  const std::string log_name = "test_plan_using_subframes";
+  ros::NodeHandle nh(log_name);
+
+  auto planning_scene_monitor = std::make_shared<planning_scene_monitor::PlanningSceneMonitor>("robot_description");
+  moveit::planning_interface::PlanningSceneInterface planning_scene_interface;
+  moveit::planning_interface::MoveGroupInterface group("panda_arm");
+  group.setPlanningTime(10.0);
+
+  spawnCollisionObjects(planning_scene_interface);
+  moveit_msgs::AttachedCollisionObject att_coll_object;
+  att_coll_object.object.id = "cylinder";
+  att_coll_object.link_name = "panda_hand";
+  att_coll_object.object.operation = att_coll_object.object.ADD;
+  planning_scene_interface.applyAttachedCollisionObject(att_coll_object);
+
+  tf2::Quaternion target_orientation;
+  target_orientation.setRPY(0, 180.0 / 180.0 * M_PI, 90.0 / 180.0 * M_PI);
+  geometry_msgs::PoseStamped target_pose_stamped;
+  target_pose_stamped.pose.orientation = tf2::toMsg(target_orientation);
+  target_pose_stamped.pose.position.z = Z_OFFSET;
+
+  ROS_INFO_STREAM_NAMED(log_name, "Moving to bottom of box with cylinder tip");
+  target_pose_stamped.header.frame_id = "box/bottom";
+  ASSERT_TRUE(moveToCartPose(target_pose_stamped, group, "cylinder/tip"));
+  planning_scene_monitor->requestPlanningSceneState();
+  {
+    planning_scene_monitor::LockedPlanningSceneRO planning_scene(planning_scene_monitor);
+
+    // get the tip and box subframe locations in world
+    Eigen::Isometry3d eef = planning_scene->getFrameTransform("cylinder/tip");
+    Eigen::Isometry3d box_subframe = planning_scene->getFrameTransform(target_pose_stamped.header.frame_id);
+    Eigen::Isometry3d target_pose;
+    tf::poseMsgToEigen(target_pose_stamped.pose, target_pose);
+
+    // expect that they are identical
+    std::stringstream ss;
+    ss << "box frame: \n" << box_subframe.matrix() << "\ncylinder frame: \n" << eef.matrix();
```

Comment:
We should print the correct transform for comparison.
```suggestion
    ss << "target frame: \n" << (box_subframe * target_pose).matrix() << "\ncylinder frame: \n" << eef.matrix();
```